.github/workflows/ci.yaml

```diff
@@ -5,28 +5,34 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['3.7', '3.8', '3.9', '3.10']
+        versions:
+          # test earliest biopython version supporting each python version
+          # from https://github.com/biopython/biopython/blob/master/NEWS.rst
+          - {python: '3.7', biopython: '1.73'}
+          - {python: '3.8', biopython: '1.76'}
+          - {python: '3.9', biopython: '1.79'}
+          - {python: '3.10', biopython: ''} # empty string = latest supported
     defaults:
       run:
         shell: bash -l {0}
     steps:
     - uses: actions/checkout@v2
     - uses: conda-incubator/setup-miniconda@v2
       with:
-        python-version: ${{ matrix.python-version }}
+        python-version: ${{ matrix.versions.python }}
         mamba-version: "*"
         channels: conda-forge,bioconda
         channel-priority: true
         activate-environment: test
-    - run: mamba install mafft raxml fasttree iqtree vcftools pip numpy
-    - run: pip install biopython==1.67
+    - run: mamba install mafft raxml fasttree iqtree vcftools
+    - run: mamba install biopython=${{ matrix.versions.biopython }}
     - run: pip install -e .[dev]
     - run: conda info
     - run: conda list
     - run: pytest -c pytest.python3.ini  --cov-report=xml --cov=augur
     - run: cram --shell=/bin/bash tests/functional/*.t tests/builds/*.t
     - run: bash tests/builds/runner.sh
-    - if: github.repository == 'nextstrain/augur' && matrix.python-version == 3.10
+    - if: github.repository == 'nextstrain/augur' && matrix.versions.python == 3.10
       uses: codecov/codecov-action@v2
       with:
         fail_ci_if_error: true
```

Comment:
It seems better to me to split the codecov step out into its own job in this workflow instead of continuing to piggyback on the `test` job and having to ensure it selects just one of the matrix runs.

(I realize this is slightly tangential to what the PR is doing, but since we're looking at it.)

Reply:
(To clarify: I don't consider this a required change for this PR. But a small improvement that'd be good to do (if it's indeed as simple as it sounds) in this PR or another one to follow.)

Reply:
Sounds good, I'll make a new PR following merge of this one.